src/core/meshcop/commissioner.cpp

```diff
@@ -297,6 +297,34 @@ otError Commissioner::AddJoiner(const Mac::ExtAddress *aEui64, const char *aPSKd
     return error;
 }
 
+otError Commissioner::GetNextJoinerInfo(uint16_t &aIterator, otJoinerInfo &aJoiner)
+{
+    otError error = OT_ERROR_NONE;
+    size_t  index;
+
+    for (index = aIterator; index < OT_ARRAY_LENGTH(mJoiners); index++)
+    {
+        if (!mJoiners[index].mValid)
+        {
+            continue;
+        }
+
+        memset(&aJoiner, 0, sizeof(aJoiner));
+
+        aJoiner.mAny = mJoiners[index].mAny;
+        memcpy(&aJoiner.mEui64, &mJoiners[index].mEui64, sizeof(aJoiner.mEui64));
+        strlcpy(aJoiner.mPsk, mJoiners[index].mPsk, sizeof(aJoiner.mPsk));
+        aJoiner.mExpirationTime = mJoiners[index].mExpirationTime;
```

Comment:
How about changing what we return in `aJoiner` to be number of `ms` till entry will be expired/removed.
``` aJoiner.mTimeout  =  mJoiners[index].mExpirationTime - TimerMilli::GetNow()```

The `mExpirationTime` in `Joiner` is the time the entry will be timed out and removed. This relative to current time from  `TimerMilli::GetNow()` (which is internal to OT and not directly available/acceeisble outside).